ethsigner/src/main/java/tech/pegasys/ethsigner/http/JsonRpcHttpService.java

```diff
@@ -98,6 +98,13 @@ public void stop(final Future<Void> stopFuture) {
         });
   }
 
+  public int actualPort() {
```

Comment:
confirming - if this is called before start, it returns 0 - otherwise the port used by the httpServer? Is this ok, should it throw? Should the concept of construction/start be atomic?

Reply:
Confirmed.

Zero is returned, unless the httpServer is not null (in theory that could also return zero, unless startup is sufficiently complete).

I'm fine with this behaviour, as Java socket use zero to denote an unassigned / unbound socket ...which is what we would have,